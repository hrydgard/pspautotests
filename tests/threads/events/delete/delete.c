#include "../sub_shared.h"

SETUP_SCHED_TEST;

#define DELETE_TEST(title, flag) { \
	int result = sceKernelDeleteEventFlag(flag); \
	if (result == 0) { \
		printf("%s: OK\n", title); \
	} else { \
		printf("%s: Failed (%X)\n", title, result); \
	} \
}

int main(int argc, char **argv) {
	SceUID flag = sceKernelCreateEventFlag("delete1", 0, 0, NULL);

	DELETE_TEST("Normal", flag);
	DELETE_TEST("NULL", 0);
	DELETE_TEST("Invalid", 0xDEADBEEF);
	DELETE_TEST("Deleted", flag);
	
	BASIC_SCHED_TEST("Delete other",
		result = sceKernelDeleteEventFlag(flag2);
	);
	BASIC_SCHED_TEST("Delete same",
		result = sceKernelDeleteEventFlag(flag1);
	);
	BASIC_SCHED_TEST("NULL",
		result = sceKernelDeleteEventFlag(0);
	);

	return 0;
	
}