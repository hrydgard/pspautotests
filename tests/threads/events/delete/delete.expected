Normal: OK
NULL: Failed (8002019A)
Invalid: Failed (8002019A)
Deleted: Failed (8002019A)
Delete other: A1B1C2E1D1F2 (thread=800201B5, thread bits=00000000, main=00000000)
Delete same: A1B1C2D1E2F1 (thread=800201B5, thread bits=00000000, main=00000000)
NULL: A1B1C2E1D1F2 (thread=800201B5, thread bits=00000000, main=8002019A)
